_source/_posts/2021-11-08-k8s-api-server-oidc.md

```diff
@@ -0,0 +1,565 @@
+---
+layout: blog_post
+title: "How to Secure Your Kubernetes Cluster with OpenID Connect and RBAC"
+author: deepu-sasidharan
+by: advocate
+communities: [kubernetes, devops, security]
+description: "Learn how to set up Okta as an OpenID Connect provider using Terraform for access to Kubernetes API server."
+tags: [kubernetes, devops, oidc, okta, terraform]
+tweets:
+  - "Use #OIDC to access #Kubernetes API server securely. Set it up using Terraform."
+  - "Access #Kubernetes API server using your favorite #OIDC provider!"
+type: conversion
+image: blog/k8s-api-server-oidc/kube-login-oidc.jpg
+---
+
+A Kubernetes (k8s) cluster comprises worker machines called nodes and a control plane consisting of the API server, scheduler, etcd, controller manager, and cloud controller manager — in case of a PaaS. The containers deployed to the cluster will run in pods on the worker nodes. At the same time, the control plane takes care of scheduling, responding to requests, and managing the cluster.
+
+{% img blog/k8s-api-server-oidc/kube-architecture.jpg alt:"K8s architecture" width:"800" %}{: .center-image }
+
+When you are communicating with a Kubernetes cluster, let's say using kubectl or a client library or a tool like [KDash](https://kdash.cli.rs/), you are primarily interacting with the [Kubernetes API server](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/). The API server is responsible for managing the cluster and is responsible for handling requests from a client.
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+# Why is it required to secure the API server?
```

Comment:
This makes me wonder "required by whom?". How about saying "why secure the API server" or "how is the API server secured" instead?